docker-storage-setup.sh

```diff
@@ -627,9 +626,18 @@ check_wipe_block_dev_sig() {
 # their absolute path for processing in rest of the script.
 canonicalize_block_devs() {
   local devs=$1 dev
+  local linktarget
   local devs_abs dev_abs
 
   for dev in ${devs}; do
+    # If the device name is a symlink, follow it and use the target
+    if [ -h "$dev" ];then
+      linktarget=$(readlink -e $dev)
+      if [ $? -ne 0 ] || [ ! -b "$linktarget" ] ; then
+        Fatal "$linktarget doesn't link to a valid block device."
```

Comment:
We already do block device checks on dev_abs so no need to do it again. How about something like.

if ! dest_dev=$(readlink -e $dev);then
    Fatal "readlink failed to resolve device path $dev"
fi